### PR TITLE
Type Cleanups, focused mostly on Slots and Particles

### PR DESCRIPTION
**Pull request by lindner**
- Add return types in many places
- Convert some types to use Readonly<>
- Make private methods private
- Remove 'use strict' from typescript
- Remove non typedoc jsdoc (@class/@method)
- Remove vestigal sourceMappingURL comments
- Add missing async to setHandles/onHandleDesync/onHandleUpdate/etc in particle.ts and fix callers
- Add missing license boilerplate

### CODE REVIEW

File: src/runtime/multiplexer-dom-particle.ts [hunk 1]
```diff
@@ -28,19 +28,24 @@ export class MultiplexerDomParticle extends TransformationDomParticle {
...
+
     for (const [connectionName, otherHandle] of handles) {
       if (skipConnectionNames.includes(connectionName)) {
         continue;
       }
       // TODO(wkorman): For items with embedded recipes we may need a map
       // (perhaps id to index) to make sure we don't map a handle into the inner
       // arc multiple times unnecessarily.
-      otherMappedHandles.push(`use '${await arc.mapHandle(otherHandle.storage)}' as v${index}`);
+
```
Comment (lindner) [inline src/runtime/multiplexer-dom-particle.ts]: There's some oddities where the types specified don't seem to match with the calling convention...  Also see above where there's a distinction between the Recipe version of Handle and the base Handle class.

fyi to @shaper who might have historical knowledge...

File: src/runtime/particle-execution-context.ts [hunk 2]
```diff
@@ -152,7 +152,7 @@ export class ParticleExecutionContext {
         return new Promise((resolve, reject) =>
           pec.apiPort.ArcMapHandle(id => {
             resolve(id);
-          }, arcId, handle));
+          }, arcId, handle));  // recipe handle vs not?
```
Comment (lindner) [inline src/runtime/particle-execution-context.ts]: Here's a weird case where the passed in Handle does not match the `ArcMapHandle` signatures in `api-channel.ts` and `particle-execution-host.ts`

It appears that the called code only relies on the handle.id, so I guess that's why it works.

File: src/runtime/slot-context.ts [hunk 3]
```diff
@@ -136,10 +136,15 @@ export class ProvidedSlotContext extends SlotContext {
...
   }
 
-  get container() { return this._container; }
-  get containerAvailable() { return !!this._container; }
+  get container(): HTMLElement {
+    return this._container;
+  }
+
+  get containerAvailable(): boolean {
+    return !!this._container;
```
Comment (sjmiles) [inline src/runtime/slot-context.ts]: Tangential (and I know you didn't write it), but personally I don't like `!!` and prefer explicit cast to Boolean.

File: src/runtime/slot-dom-consumer.ts [hunk 4]
```diff
@@ -21,7 +21,8 @@ export interface DomRendering extends Rendering {
   liveDom?: Template;
 }
 
-const templateByName = new Map();
+// tslint:disable-next-line: no-any
+const templateByName = new Map<string, any>();
```
Comment (sjmiles) [inline src/runtime/slot-dom-consumer.ts]: This is a map of `string` to `Template` (the latter is a DOM construct).

**Comment by lindner** [comment]
Removed some primitive return types where it is obvious.  I think we may want to revisit our thinking about letting typescript infer, especially for class methods that get inherited.

You can see in this PR that return types caught async vs not usage.

File: src/runtime/slot-dom-consumer.ts (hunk 4 above)
Comment (lindner) [reply src/runtime/slot-dom-consumer.ts]: Thanks

File: src/runtime/slot-context.ts (hunk 3 above)
Comment (lindner) [reply src/runtime/slot-context.ts]: Started a topic in chat about using https://eslint.org/docs/rules/no-implicit-coercion

**Comment by lindner** [comment]
Also of note: increases type coverage 0.5%

**Comment by lindner** [comment]
Filed #2970 and #2969 for issues uncovered in this PR.

**Pull request merged**